agent/lib/kontena/workers/container_info_worker.rb

```diff
@@ -38,22 +44,26 @@ def on_container_event(topic, event)
 
       container = Docker::Container.get(event.id)
       if container
-        self.publish_info(container)
-        self.notify_coroner(container) if container.suspiciously_dead?
+        publish_info(container)
       end
     rescue Docker::Error::NotFoundError
-      self.publish_destroy_event(event)
+      publish_destroy_event(event)
     rescue => exc
       error "#{exc.class.name}: #{exc.message}"
       error exc.backtrace.join("\n")
     end
 
     def on_container_publish_info(topic, container)
-      self.publish_info(container)
+      publish_info(container)
     end
 
     def on_websocket_connected(topic, data)
-      self.publish_all_containers
+      publish_all_containers
+      container_coroner.start
```

Comment:
Could this be missed in some cases? For example if `ContainerInfoWorker` crashes for some random reason, when it restarts it won't see the `websocket:connected` event at all and thus the coroner never gets started.